src/pseudopeople/utilities.py

```diff
@@ -192,6 +192,35 @@ def cleanse_integer_columns(column: pd.Series) -> pd.Series:
     return column
 
 
+def number_of_tokens_per_string(s1, s2):
+    """
+    Calculates the number of tokens in each string of a series.
+    s1 is a pd.Series of tokens and we want to see how many tokens exist in each
+    string of s2.
+    """
+
+    s2 = s2.astype(str)
+    strings = s2.unique()
+    tokens_per_string = pd.Series(
+        (sum(occurrences(s, str(token)) for token in s1) for s in strings), index=strings
```

Comment:
Would this work or is `occurrences` doing something fancier?

`tokens_per_string = pd.Series(sum((s.count(str(token)) for token in s1) for s in strings), index=strings)`

Reply:
Fancier -- .count doesn't work with overlapping substrings, i.e.

```python-console
>>> 'eveve'.count('eve')
1
```

Reply:
That would work but yes but we map it back to the length of the original series that goes in because for tests we eventually take the mean so we need to preserve the number of times each string occurs.